reconcile/openshift_prometheusrules.py

```diff
@@ -0,0 +1,275 @@
+import logging
+import semver
+import json
+import subprocess
+import os
+import templates
+import jinja2
+
+import utils.gql as gql
+import utils.threaded as threaded
+import reconcile.openshift_base as ob
+
+from utils.defer import defer
+from utils.openshift_resource import (OpenshiftResource as OR,
+                                      ResourceInventory)
+
+
+class Jinja2TemplateError(Exception):
+    def __init__(self, msg):
+        super(Jinja2TemplateError, self).__init__(
+            "error processing jinja2 template: " + str(msg)
+        )
+
+
+NAMESPACES_QUERY = """
+{
+  performance_parameters_v1 {
+    labels
+    name
+    component
+    prometheusLabels
+    selectorJob
+    additionalSelectors
+    namespaces {
+      namespace {
+        name
+      }
+      overrideSelectorJob
+      overrideAdditionalSelectors
+    }
+    app {
+      name
+      namespaces {
+        name
+        cluster {
+          observabilityNamespace {
+            name
+            cluster {
+              name
+              serverUrl
+              jumpHost {
+                hostname
+                knownHosts
+                user
+                port
+                identity {
+                  path
+                  field
+                  format
+                }
+              }
+              automationToken {
+                path
+                field
+                format
+              }
+              disable {
+                integrations
+              }
+            }
+          }
+          name
+          serverUrl
+          jumpHost {
+            hostname
+            knownHosts
+            user
+            port
+            identity {
+              path
+              field
+              format
+            }
+          }
+          automationToken {
+            path
+            field
+            format
+          }
+          disable {
+            integrations
+          }
+        }
+      }
+    }
+    availability {
+      kind
+      metric
+      errorBudget
+    }
+    latency {
+      kind
+      metric
+      threshold
+      percentile
+    }
+    rawRecording {
+      record
+      expr
+      labels
+    }
+    rawAlerting {
+      alert
+      expr
+      for
+      labels
+      annotations {
+        message
+        runbook
+        dashboard
+        link_url
+      }
+    }
+  }
+}
+"""
+
+QONTRACT_INTEGRATION = 'openshift-prometheusrules'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def fetch_desired_state(results, ri, temp_dir_path):
+    os.chdir(temp_dir_path)
+    for pp in results['performance_parameters_v1']:
+        componentName = pp['component']
+        for namespace_info in pp['app']['namespaces']:
+            namespace = namespace_info['name']
+            cluster = namespace_info['cluster']
+            observabilityNamespace = cluster['observabilityNamespace']
```

Comment:
the observabilityNamespace wasn't required, was it? if it's not required, better to do: `cluster.get('observabilityNamespace')`

Reply:
Ack, changing